object.go

```diff
@@ -58,7 +63,16 @@ func set(o *Object, key string, idx uint32, val interface{}) error {
 		return nil
 	}
 
-	C.ObjectSetIdx(o.ptr, C.uint32_t(idx), value.ptr)
+	if internal {
+		inserted := C.ObjectSetInternal(o.ptr, C.uint32_t(idx), value.ptr)
+
+		if inserted == 0 {
+			return errors.New("v8go: index exceeded internal field count")
```

Comment:
These fields are internal to host/go code, so getting and setting them should be more like getting and setting a go slice.  An index out of range error in Go is reported as a panic, so I think we should do the same here.  This is because the error isn't something that we expect the caller to handle, but more an indication of a bug in the caller. 
 Likely the caller would be using these fields more like struct field anyways.

If the library user happens to be using these internal fields as a bounded array, then we could expose `InternalFieldCount` so the assignment can be conditional (similar using the `len` function on a slice)